roscpp_azure_iothub/src/ros_azure_iothub_cpp_node.cpp

```diff
@@ -408,18 +410,26 @@ static void deviceTwinCallback(DEVICE_TWIN_UPDATE_STATE update_state, const unsi
 
 static bool ReadKeyFromFile(const std::string fileName, std::string &key)
 {
-    //key = NULL;
+    key = "";
     if (fileName.empty())
     {
         ROS_ERROR("File name empty.");
         return false;
     }
 
     std::ifstream f(fileName); //taking file as inputstream
```

Comment:
will this throw on an invalid file?

Reply:
If the path is invalid or the file doesn't exist, it won't throw. ifstream f will be false and it will go down the "Could not read from file." error path.

If the file is invalid because it's not the correct key or not a proper key at all, that will get validated by IoTHubDeviceClient_SetOption. The proper key validation is likely pretty involved so I figured it shouldn't be duplicated.